src/components/manifold-subscription-details/manifold-subscription-details.tsx

```diff
@@ -44,12 +47,16 @@ export class ManifoldSubscriptionCreate {
   async componentWillLoad() {
     await customElements.whenDefined('manifold-init');
     const core = document.querySelector('manifold-init') as HTMLManifoldInitElement;
-    this.connection = await core.initialize({
+
+    const connection = await core.initialize({
       element: this.el,
       componentVersion: '<@NPM_PACKAGE_VERSION@>',
       version: 0,
     });
 
+    state.connection = connection;
+    this.connection = connection;
```

Comment:
this might not be the best way to get the connection into the store. still figuring out how stores work here. make suggesitons if you have them

Reply:
Yeah, the connection isn't really _state_ to begin with, right? And different components need different instances of the connection for reporting purposes. So if we ever had a scenario where two components from the same library were on the same page, then whichever one is the last to initialize would overwrite whatever was in the store previously, and then all components would be reporting as the same component.

Reply:
If you see my latest comment, I mention using a store factory and creating multiple stores, one for each component

Reply:
That seems like an improvement. So if one web component creates a store and then renders functional components, how do those functional components access that store? Would you pass the _name_ of that store to the functional component so that they always know which one to reference?

Reply:
there's a few different options I think. I'll just play around with it until I find an approach that works well